Project/DAQ_System/Program/Src/Platform/I2C/STM/i2cHalWrapper_stmf4.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * i2cHalWrapper_stmf4.cpp
+ *
+ *  Created on: Mar 4, 2024
+ *      Author: Nico
+ */
+#include "i2cHalWrapper_stmf4.hpp"
+
+namespace platform {
+
+	i2cHalWrapperStmf4::i2cHalWrapperStmf4(I2C_HandleTypeDef &hi2c, uint8_t SAD)
+			: bx_i2c_(hi2c) {
+		slave_add_ = SAD;
+	}
+
+	i2cHalWrapperStmf4::~i2cHalWrapperStmf4(){}
+
+	void i2cHalWrapperStmf4::Master_Transmit(uint8_t *commands){
+		HAL_I2C_Master_Transmit(&bx_i2c_,slave_add_, commands, kMaxBytes, 0xFFFFFFFF);
+	}
+	void i2cHalWrapperStmf4::Master_Receive(uint8_t *response){
+		HAL_I2C_Master_Receive(&bx_i2c_, slave_add_, response, response_size_, 0xFFFFFFFF);
+	}
```

Comment:
I think the address variable could be an argument of `Transmit()` and `Receive()`. An I2C line should be able to support multiple devices — multiple addresses — and therefore should not depend on exactly one address via the constructor.